README.md
Install NPM or Yarn

npm install or yarn install


CHANGELOG.md
## [1.2.0] 2021-01-26
### Bug fixing
- https://github.com/creativetimofficial/argon-dashboard-react/pull/38
- https://github.com/creativetimofficial/argon-dashboard-react/pull/33
- https://github.com/creativetimofficial/argon-dashboard-react/issues/29
- https://github.com/creativetimofficial/argon-dashboard-react/issues/23
### Major style changes
- Moved all `src/scss/argon-dashboard/bootstrap` files to `node_modules/bootstrap`
### Deleted components
### Added components
### Deleted dependencies
- @types/googlemaps (we'll use simple Google Maps API with Vanilla JS instead)
- @types/markerclustererplus (we'll use simple Google Maps API with Vanilla JS instead)
- @types/react (we'll use simple Google Maps API with Vanilla JS instead)
- react-google-maps (we'll use simple Google Maps API with Vanilla JS instead)
### Added dependencies
+ bootstrap@4.6.0 (so that we can easily update bootstrap version)
+ jquery@3.5.1 (since bootstrap requires it - we are not using, we are only using bootstrap scss)
+ node-sass-package-importer@5.3.2 (so that we can import bootstrap scss from `node_modules`)
### Updated dependencies
```
@fortawesome/fontawesome-free    5.12.1   →   5.15.2
chart.js                          2.9.3   →    2.9.4
moment                           2.24.0   →   2.29.1
node-sass                        4.13.1   →   4.14.1
nouislider                       14.1.1   →   14.6.3
react                           16.12.0   →   17.0.1
react-chartjs-2                   2.9.0   →   2.11.1
react-copy-to-clipboard           5.0.2   →    5.0.3
react-datetime                   2.16.3   →    3.0.4
react-dom                       16.12.0   →   17.0.1
react-router-dom                  5.1.2   →    5.2.0
react-scripts                    ^3.4.0   →    4.0.1
reactstrap                        8.4.1   →    8.9.0
eslint-plugin-flowtype           3.13.0   →    5.2.0
typescript                        3.7.5   →    4.1.3
```
### Warning
**The TypeScript and jQuery dependencies are installed only to stop console warnings on install. They are not actually used in our product. So the product is not based on TypeScript or jQuery!**
_The following warnings will appear when running the installation command, but they do not affect the UI or the functionality of the product (they will be solved in our next update):_
```
npm WARN react-datetime@3.0.4 requires a peer of react@^16.5.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-popper@1.3.7 requires a peer of react@0.14.x || ^15.0.0 || ^16.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN create-react-context@0.3.0 requires a peer of react@^0.14.0 || ^15.0.0 || ^16.0.0 but none is installed. You must install peer dependencies yourself.
```
_If they will persist in our 2.*.* version, we will drop their usages and replace them with other plugins._
_In development mode, some of the above plugins will throw a warning because they still use React v16 syntax. If the error will persist in our 2.*.* version, we will drop their usage and replace them with other plugins._

## [1.1.0] 2020-02-10
### Bug fixing
- https://github.com/creativetimofficial/argon-dashboard-react/issues/18 (Added this in live docs, we need for our product to have a homepage prop inside the package.json. If build is not working, just delete the homepage prop from inside the package.json, or configure it to your own specs.)
- https://github.com/creativetimofficial/argon-dashboard-react/issues/17
- https://github.com/creativetimofficial/argon-dashboard-react/issues/15
- https://github.com/creativetimofficial/argon-dashboard-react/issues/12
- https://github.com/creativetimofficial/argon-dashboard-react/issues/3
- Changed `componentWillMount` with `constructor` function inside `src/views/Index.js` (this was for initializing the charts)
### Major style changes
- Changed the whole `src/assets/scss/*` folder
- Changed the whole `src/assets/css/*` folder
### Deleted components
### Added components
### Deleted dependencies
### Added dependencies
+ eslint-plugin-flowtype@3.13.0 (To stop following Warning: `npm **WARN** eslint-config-react-app@5.2.0 requires a peer of eslint-plugin-flowtype@3.x but none is installed. You must install peer dependencies yourself.`)
+ @fortawesome/fontawesome-free@5.12.1 (Easier to maintain as a dependency, rather than having to download new versions)
+ gulp@4.0.2 (For licenses copyrights)
+ gulp-append-prepend@1.0.8 (For licenses copyrights)
### Updated dependencies
```
chart.js                     2.7.3   →     2.9.3
node-sass                   4.11.0   →    4.13.1
nouislider                  13.1.1   →    14.1.1
react                       16.8.4   →   16.12.0
react-chartjs-2              2.7.4   →     2.9.0
react-copy-to-clipboard      5.0.1   →     5.0.2
react-dom                   16.8.4   →   16.12.0
react-router-dom             4.3.1   →     5.1.2
react-scripts                2.1.8   →     3.3.1
reactstrap                   7.1.0   →     8.4.1
@types/googlemaps          3.30.18   →    3.39.2
@types/react                16.8.7   →   16.9.19
typescript                3.3.3333   →     3.7.5
```
### Warning
** The following warnings could not be solved, due to some of our dependencies, however, they do not affect the functionality or the UI of the product: **
```
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated popper.js@1.16.1: Popper changed home, find its new releases at @popperjs/core
```
** The following warning could not be solved due to the usage of `react-google-maps` (Note: if the warning will persist after React changes version to 17, we'll drop the support for `react-google-maps` and replace them with other react library for maps):**
```

src/views/Constants.js
const production = {
    DSD: [

        {
            id: 1,
            ds_name: 'Koralai Pattu North',
            ds_id: 'kp_north',

        },
        {
            id: 2,
            ds_name: 'Koralai Pattu West',
            ds_id: 'kp_west',

        },
        {
            id: 3,
            ds_name: 'Koralai Pattu',
            ds_id: 'kp',

        },

        {
            id: 4,
            ds_name: 'Eravur Pattu',
            ds_id: 'ep',

        },
        {
            id: 5,
            ds_name: 'Eravur Town',
            ds_id: 'et',

        },
        {
            id: 6,
            ds_name: 'Manmunai North',
            ds_id: 'mm_north',

        },
        {
            id: 7,
            ds_name: 'Kattankudy',
            ds_id: 'kkdy',

        },
        {
            id: 8,
            ds_name: 'Manmunai Pattu',
            ds_id: 'mm_pattru',

        },
        {
            id: 9,
            ds_name: 'Manmunai South‐West',
            ds_id: 'mm_sw',

        },
        {
            id: 10,
            ds_name: 'Poratheevu Pattu',
            ds_id: 'pt_pattru',

        },
        {
            id: 11,
            ds_name: 'Manmunai South & Eruvil Pattu',
            ds_id: 'mm_south_ep',

        },
        {
            id: 12,
            ds_name: 'Koralai Pattu Central',
            ds_id: 'kp_central',

        },
        {
            id: 13,
            ds_name: 'Koralai Pattu South',
            ds_id: 'kp_south',

        },
        {
            id: 14,
            ds_name: 'Manmunai West',
            ds_id: 'mm_west',

        }
    ],
    GND: [{
            id: 1,
            name: 'Mankerny Central',
            gn_id_readable: '211',
            gn_id: '211',
            ds_name: 'Koralai Pattu North',
            ds_id: 'kp_north',
        },
        {
            id: 2,
            name: 'Kayankerni',
            gn_id_readable: '211 A',
            gn_id: '211_A',
            ds_name: 'Koralai Pattu North',
            ds_id: 'kp_north',
        },

        {
            id: 3,
            name: 'Punanai East',
            gn_id_readable: '211 B',
            gn_id: '211_B',
            ds_name: 'Koralai Pattu North',
            ds_id: 'kp_north',
        },

        {
            id: 4,
            name: 'Panichankerni',
            gn_id_readable: '211 C',
            gn_id: '211_C',
            ds_name: 'Koralai Pattu North',
            ds_id: 'kp_north',
        },

        {
            id: 5,
            name: 'Vattavan',
            gn_id_readable: '211 D',
            gn_id: '211_D',
            ds_name: 'Koralai Pattu North',
            ds_id: 'kp_north',
        },

        {
            id: 6,
            name: 'Meeravodai Muslim West',
            gn_id_readable: '207',
            gn_id: '207',
            ds_name: 'Koralai Pattu West',
            ds_id: 'kp_west',
        },

        {
            id: 7,
            name: 'Mancholai',
            gn_id_readable: '207 A',
            gn_id: '207_A',
            ds_name: 'Koralai Pattu West',
            ds_id: 'kp_west',
        },

        {
            id: 8,
            name: 'Meeravodai Muslim East',
            gn_id_readable: '207 B',
            gn_id: '207_B',
            ds_name: 'Koralai Pattu West',
            ds_id: 'kp_west',
        },

        {
            id: 9,
            name: 'Oddamavady',
            gn_id_readable: '208',
            gn_id: '208',
            ds_name: 'Koralai Pattu West',
            ds_id: 'kp_west',
        },

        {
            id: 10,
            name: 'Paper Town',
            gn_id_readable: '210',
            gn_id: '210',
            ds_name: 'Koralai Pattu West',
            ds_id: 'kp_west',
        },

        {
            id: 11,
            name: 'Meeravodai Tamil',
            gn_id_readable: '202',
            gn_id: '202',
            ds_name: 'Koralai Pattu',
            ds_id: 'kp',
        },

        {
            id: 12,
            name: 'Karuwakerny',
            gn_id_readable: '202 A',
            gn_id: '202_A',
            ds_name: 'Koralai Pattu',
            ds_id: 'kp',
        },

        {
            id: 13,
            name: 'Karuwakerny',
            gn_id_readable: '202 A',
            gn_id: '202_A',
            ds_name: 'Koralai Pattu',
            ds_id: 'kp',
        },

        {
            id: 14,
            name: 'Kinnayadi',
            gn_id_readable: '202 B',
            gn_id: '202_B',
            ds_name: 'Koralai Pattu',
            ds_id: 'kp',
        },

        {
            id: 15,
            name: 'Sungankerni',
            gn_id_readable: '202 C',
            gn_id: '202_C',
            ds_name: 'Koralai Pattu',
            ds_id: 'kp',
        },


        {
            id: 16,
            name: 'Periyapullumalai',
            gn_id_readable: '146',
            gn_id: '146',
            ds_name: 'Eravur Pattu',
            ds_id: 'ep',
        },

        {
            id: 17,
            name: 'koppaveli',
            gn_id_readable: '146 A',
            gn_id: '146_A',
            ds_name: 'Eravur Pattu',
            ds_id: 'ep',
        },

        {
            id: 18,
            name: 'Rugam',
            gn_id_readable: '185',
            gn_id: '185',
            ds_name: 'Eravur Pattu',
            ds_id: 'ep',
        },

        {
            id: 19,
            name: 'Karadiyanaru',
            gn_id_readable: '185 A',
            gn_id: '185_A',
            ds_name: 'Eravur Pattu',
            ds_id: 'ep',
        },

        {
            id: 20,
            name: 'Kithulwewa',
            gn_id_readable: '185 B',
            gn_id: '185_B',
            ds_name: 'Eravur Pattu',
            ds_id: 'ep',
        },

        {
            id: 21,
            name: 'Eravur 1',
            gn_id_readable: '190',
            gn_id: '190',
            ds_name: 'Eravur Town',
            ds_id: 'et',
        },

        {
            id: 22,
            name: 'Eravur 1A',
            gn_id_readable: '190 A',
            gn_id: '190 A',
            ds_name: 'Eravur Town',
            ds_id: 'et',
        },

        {
            id: 23,
            name: 'Eravur 1B',
            gn_id_readable: '190 B',
            gn_id: '190 B',
            ds_name: 'Eravur Town',
            ds_id: 'et',
        },

        {
            id: 24,
            name: 'Eravur 2',
            gn_id_readable: '191',
            gn_id: '191',
            ds_name: 'Eravur Town',
            ds_id: 'et',
        },

        {
            id: 25,
            name: 'Eravur 2A',
            gn_id_readable: '191 A',
            gn_id: '191_A',
            ds_name: 'Eravur Town',
            ds_id: 'et',
        },

        {
            id: 26,
            name: 'Manchanthoduwai North',
            gn_id_readable: '168',
            gn_id: '168',
            ds_name: 'Manmunai North',
            ds_id: 'mm_north',
        },

        {
            id: 27,
            name: 'Manchanthoduwai South',
            gn_id_readable: '168 A',
            gn_id: '168_A',
            ds_name: 'Manmunai North',
            ds_id: 'mm_north',
        },

        {
            id: 28,
            name: 'Navatkudah',
            gn_id_readable: '168 B',
            gn_id: '168_B',
            ds_name: 'Manmunai North',
            ds_id: 'mm_north',
        },

        {
            id: 29,
            name: 'Navatkudah East',
            gn_id_readable: '168 C',
            gn_id: '168_C',
            ds_name: 'Manmunai North',
            ds_id: 'mm_north',
        },

        {
            id: 29,
            name: 'Navatkudah South',
            gn_id_readable: '168 D',
            gn_id: '168_D',
            ds_name: 'Manmunai North',
            ds_id: 'mm_north',
        },

        {
            id: 30,
            name: 'Nochimunai',
            gn_id_readable: '171 A',
            gn_id: '171_A',
            ds_name: 'Manmunai North',
            ds_id: 'mm_north',
        },

        {
            id: 31,
            name: 'Kathankudy 5',
            gn_id_readable: '164 A',
            gn_id: '164_A',
            ds_name: 'Kattankudy',
            ds_id: 'kkdy',
        },

        {
            id: 32,
            name: 'Kathankudy 3',
            gn_id_readable: '165',
            gn_id: '165',
            ds_name: 'Kattankudy',
            ds_id: 'kkdy',
        },

        {
            id: 33,
            name: 'New Kathankudy South',
            gn_id_readable: '167 C',
            gn_id: '167_C',
            ds_name: 'Kattankudy',
            ds_id: 'kkdy',
        },

        {
            id: 34,
            name: 'Kathankudy 6',
            gn_id_readable: '162',
            gn_id: '162',
            ds_name: 'Kattankudy',
            ds_id: 'kkdy',
        },

        {
            id: 35,
            name: 'Kathankudy 2 North',
            gn_id_readable: '166 A',
            gn_id: '166_A',
            ds_name: 'Kattankudy',
            ds_id: 'kkdy',
        },

        {
            id: 36,
            name: 'Arayampathy Central',
            gn_id_readable: '156 D',
            gn_id: '156_D',
            ds_name: 'Manmunai Pattu',
            ds_id: 'mm_pattru',
        },

        {
            id: 37,
            name: 'Kankeyanodai South',
            gn_id_readable: '155 B',
            gn_id: '155_B',
            ds_name: 'Manmunai Pattu',
            ds_id: 'mm_pattru',
        },

        {
            id: 38,
            name: 'Thalankuda',
            gn_id_readable: '153',
            gn_id: '153',
            ds_name: 'Manmunai Pattu',
            ds_id: 'mm_pattru',
        },

        {
            id: 39,
            name: 'Mavilankanthurai',
            gn_id_readable: '154',
            gn_id: '154',
            ds_name: 'Manmunai Pattu',
            ds_id: 'mm_pattru',
        },

        {
            id: 40,
            name: 'Kovilkulam',
            gn_id_readable: '156 C',
            gn_id: '156_C',
            ds_name: 'Manmunai Pattu',
            ds_id: 'mm_pattru',
        },

        {
            id: 41,
            name: 'Ampilanthurai',
            gn_id_readable: '128',
            gn_id: '128',
            ds_name: 'Manmunai South‐West',
            ds_id: 'mm_sw',
        },

        {
            id: 42,
            name: 'Ampilanthurai West',
            gn_id_readable: '128 A',
            gn_id: '128_A',
            ds_name: 'Manmunai South‐West',
            ds_id: 'mm_sw',
        },

        {
            id: 43,
            name: 'Ampilanthurai North',
            gn_id_readable: '128 B',
            gn_id: '128_B',
            ds_name: 'Manmunai South‐West',
            ds_id: 'mm_sw',
        },

        {
            id: 44,
            name: 'Katchenai',
            gn_id_readable: '128 C',
            gn_id: '128_C',
            ds_name: 'Manmunai South‐West',
            ds_id: 'mm_sw',
        },

        {
            id: 45,
            name: 'Arasadithivu',
            gn_id_readable: '129',
            gn_id: '129',
            ds_name: 'Manmunai South‐West',
            ds_id: 'mm_sw',
        },

        {
            id: 46,
            name: 'Mandur 3',
            gn_id_readable: '96',
            gn_id: '96',
            ds_name: 'Poratheevu Pattu',
            ds_id: 'pp',
        },

        {
            id: 47,
            name: 'Mandur Kottamunai',
            gn_id_readable: '96 A',
            gn_id: '96_A',
            ds_name: 'Poratheevu Pattu',
            ds_id: 'pp',
        },

        {
            id: 48,
            name: 'Palamunai',
            gn_id_readable: '97 A',
            gn_id: '97_A',
            ds_name: 'Poratheevu Pattu',
            ds_id: 'pp',
        },

        {
            id: 49,
            name: 'Mandur South',
            gn_id_readable: '97 B',
            gn_id: '97_B',
            ds_name: 'Poratheevu Pattu',
            ds_id: 'pp',
        },

        {
            id: 50,
            name: 'Thambalawattha',
            gn_id_readable: '98',
            gn_id: '98',
            ds_name: 'Poratheevu Pattu',
            ds_id: 'pp',
        },

        {
            id: 51,
            name: 'Kurukkalmadam North',
            gn_id_readable: '126 C',
            gn_id: '126_C',
            ds_name: 'Manmunai South & Eruvil Pattu',
            ds_id: 'ms_ep',
        },

        {
            id: 52,
            name: 'Kurukkalmadam South',
            gn_id_readable: '126',
            gn_id: '126',
            ds_name: 'Manmunai South & Eruvil Pattu',
            ds_id: 'ms_ep',
        },

        {
            id: 53,
            name: 'Mankadu',
            gn_id_readable: '123',
            gn_id: '123',
            ds_name: 'Manmunai South & Eruvil Pattu',
            ds_id: 'ms_ep',
        },

        {
            id: 54,
            name: 'Thethatheevu North',
            gn_id_readable: '122',
            gn_id: '122',
            ds_name: 'Manmunai South & Eruvil Pattu',
            ds_id: 'ms_ep',
        },

        {
            id: 55,
            name: 'Kaluthawalai 1',
            gn_id_readable: '119',
            gn_id: '119',
            ds_name: 'Manmunai South & Eruvil Pattu',
            ds_id: 'ms_ep',
        },

        {
            id: 56,
            name: 'Valaichenai 4 Muslim',
            gn_id_readable: '206',
            gn_id: '206',
            ds_name: 'Koralai Pattu Central',
            ds_id: 'kp_central',
        },

        {
            id: 57,
            name: 'Brainthuraichenai North',
            gn_id_readable: '206 A',
            gn_id: '206_A',
            ds_name: 'Koralai Pattu Central',
            ds_id: 'kp_central',
        },

        {
            id: 58,
            name: 'Valaichenai 5 Muslim',
            gn_id_readable: '206 B',
            gn_id: '206_B',
            ds_name: 'Koralai Pattu Central',
            ds_id: 'kp_central',
        },

        {
            id: 59,
            name: 'Brainthuraichenai South',
            gn_id_readable: '206 C',
            gn_id: '206_C',
            ds_name: 'Koralai Pattu Central',
            ds_id: 'kp_central',
        },

        {
            id: 60,
            name: 'Mavadichenai',
            gn_id_readable: '208 A',
            gn_id: '208_A',
            ds_name: 'Koralai Pattu Central',
            ds_id: 'kp_central',
        },

        {
            id: 61,
            name: 'Murakkottanchenai',
            gn_id_readable: '199',
            gn_id: '199',
            ds_name: 'Koralai Pattu South',
            ds_id: 'kp_south',
        },

        {
            id: 62,
            name: 'Thevapuram',
            gn_id_readable: '199 A',
            gn_id: '199_A',
            ds_name: 'Koralai Pattu South',
            ds_id: 'kp_south',
        },

        {
            id: 63,
            name: 'Santhiveli',
            gn_id_readable: '200',
            gn_id: '200',
            ds_name: 'Koralai Pattu South',
            ds_id: 'kp_south',
        },

        {
            id: 64,
            name: 'Thihilivaddai',
            gn_id_readable: '200 A',
            gn_id: '200_A',
            ds_name: 'Koralai Pattu South',
            ds_id: 'kp_south',
        },

        {
            id: 65,
            name: 'Palayadithona',
            gn_id_readable: '201',
            gn_id: '201',
            ds_name: 'Koralai Pattu South',
            ds_id: 'kp_south',
        },

        {
            id: 66,
            name: 'Iluppadichenai',
            gn_id_readable: '148',
            gn_id: '148',
            ds_name: 'Manmunai West',
            ds_id: 'mm_west',
        },

        {
            id: 67,
            name: 'Pavatkodichenai',
            gn_id_readable: '148 A',
            gn_id: '148_A',
            ds_name: 'Manmunai West',
            ds_id: 'mm_west',
        },

        {
            id: 68,
            name: 'Panchenai',
            gn_id_readable: '148 B',
            gn_id: '148_B',
            ds_name: 'Manmunai West',
            ds_id: 'mm_west',
        },

        {
            id: 69,
            name: 'Kanthinagar',
            gn_id_readable: '148 C',
            gn_id: '148_C',
            ds_name: 'Manmunai West',
            ds_id: 'mm_west',
        },

        {
            id: 70,
            name: 'Kanchirankudah',
            gn_id_readable: '148 D',
            gn_id: '148_D',
            ds_name: 'Manmunai West',
            ds_id: 'mm_west',
        }
    ]
};



export const config = production;